lasagne/tests/layers/test_noise.py

```diff
@@ -48,6 +48,12 @@ def test_get_output_for_no_rescale(self, layer_no_rescale):
         assert 0.4 < result_eval.mean() < 0.6
         assert (numpy.unique(result_eval) == [0., 1.]).all()
 
+    def test_get_output_for_no_rescale_dtype(self, layer_no_rescale):
+        input = theano.shared(numpy.ones((100, 100), dtype=numpy.int32))
+        result = layer_no_rescale.get_output_for(input)
+        result_eval = result.eval()
+        assert result_eval.dtype == input.dtype
```

Comment:
It would be enough to do:
```
assert result.dtype == input.dtype
```
This avoids the `eval()` step which requires a compilation. Theano already keeps track of dtypes in the graph fine.